client/matchmaking/admin-matchmaking-times.tsx

```diff
@@ -0,0 +1,690 @@
+        />
+      )}
+      <FutureTimesContainer>
+        {Array.from(futureTimesIds).map(id => {
+          const time = futureTimesById.get(id)!
+          return (
+            <HistoryCard key={time.id}>
+              <HistoryInfo>
+                <HistoryDate>{longTimestamp.format(time.startDate)}</HistoryDate>
+                <HistoryStatus>
+                  <MatchmakingStatus isEnabled={time.enabled} />
+                </HistoryStatus>
+              </HistoryInfo>
+              <HistoryActions>
+                <TextButton
+                  label='Delete'
+                  color='accent'
+                  onClick={() => onDeleteMatchmakingTime(time.id)}
+                />
+              </HistoryActions>
+            </HistoryCard>
+          )
+        })}
+      </FutureTimesContainer>
+    </>
+  )
+}
+
+function CurrentMatchmakingTime({ activeTab }: { activeTab: MatchmakingType }) {
+  const dispatch = useAppDispatch()
+
+  const [currentTime, setCurrentTime] = useState<MatchmakingTimeJson | undefined>(undefined)
+
+  const [isLoadingCurrentTime, setIsLoadingCurrentTime] = useState(false)
+  const [lastError, setLastError] = useState<Error | undefined>(undefined)
+
+  useEffect(() => {
+    const abortController = new AbortController()
+
+    setIsLoadingCurrentTime(true)
+
+    dispatch(
+      getCurrentMatchmakingTime(activeTab, {
+        signal: abortController.signal,
+        onSuccess: data => {
+          setCurrentTime(data)
+          setIsLoadingCurrentTime(false)
+          setLastError(undefined)
+        },
+        onError: err => {
+          setIsLoadingCurrentTime(false)
+          setLastError(err)
+        },
+      }),
+    )
+
+    return () => abortController.abort()
+  }, [activeTab, dispatch, setCurrentTime, setLastError])
+
+  if (isLoadingCurrentTime) {
+    return <LoadingDotsArea />
+  }
+
+  if (lastError) {
+    return <ErrorText>Error retrieving current matchmaking time: {lastError.message}</ErrorText>
+  }
+
+  if (!currentTime) {
+    return (
+      <EmptyStateContainer>
+        <EmptyStateIcon icon='schedule' />
+        <EmptyStateText>
+          No matchmaking times have been scheduled for this matchmaking type yet.
+        </EmptyStateText>
+      </EmptyStateContainer>
+    )
+  }
+
+  return (
+    <>
+      <SectionTitle>Current Matchmaking Time</SectionTitle>
+      <CurrentMatchmakingCard>
+        <HistoryInfo>
+          <HistoryDate>
+            {longTimestamp.format(currentTime.startDate)} <CurrentText>(Current)</CurrentText>
+          </HistoryDate>
+          <HistoryStatus>
+            <MatchmakingStatus isEnabled={currentTime.enabled} />
+          </HistoryStatus>
+        </HistoryInfo>
+      </CurrentMatchmakingCard>
+    </>
+  )
+}
+
+function PastMatchmakingTimes({ activeTab }: { activeTab: MatchmakingType }) {
+  const dispatch = useAppDispatch()
+
+  const [pastTimesIds, setPastTimesIds] = useImmerState<Set<number>>(new Set())
```

Comment:
Same as for the future ones, don't really see the value of this here

Reply:
I was moving the past times to an array since we don't have the same issue we have with future times here, but I've noticed another issue with using arrays or sets of objects. Which is that I get duplicated first page of past times on initial load, since the effect that's being used to load the first page of past times is being called twice in strict mode (only on dev I guess).

Now I know that React does this to surface any bugs in effects caused by calling the effect twice, but I really don't know how we're supposed to do this very basic functionality of _load some data when component mounts_ otherwise?

Reply:
Strict mode does this to surface bugs that happen when effects get called twice *which can happen in production*. There is no concept of "effects that only run on mount", all effects can run multiple times. If your code assumes it will only run on mount ever, the code is broken